internal/proxy/oauthproxy.go

```diff
@@ -185,8 +185,8 @@ func (t *upstreamTransport) getTransport() *http.Transport {
 		t.transport = &http.Transport{
 			Proxy: http.ProxyFromEnvironment,
 			DialContext: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
+				Timeout:   300 * time.Second,
```

Comment:
We should figure out a way to make this configurable. A default of 300 seconds (much like 30 seconds) is unlikely to be sufficient for everyone's needs.